templates/solaris_install.sh.erb

```diff
@@ -1,8 +1,7 @@
 function start_service() {
   service="${1:?}"
-  /opt/puppetlabs/bin/puppet resource service "${service:?}" ensure=running enable=true
+  /opt/puppetlabs/bin/puppet resource service "${service:?}" ensure=running restart=true
```

Comment:
I think you've misunderstood what the `restart` option does. https://puppet.com/docs/puppet/6.4/types/service.html#service-attribute-restart

In this case nothing happens. If there were a **refresh event**, it would attempt to restart using the command `true` (which would do nothing, but do it successfully), but I don't think there's a way to trigger a refresh event using the `puppet resource` command. I think this has to be `puppet resource service ... ensure=stopped; puppet resource service ... ensure=running`.

I'd be tempted to directly use the Solaris service management commands because we use this script for a single platform (Solaris 10).

Reply:
I think the only open question I had was whether something's shown we need to actually restart the service rather than ensure running. Is it always stopped after a package upgrade?